src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DraftReferralService.kt

```diff
@@ -487,6 +498,78 @@ class DraftReferralService(
     }
   }
 
+  private fun verifyCustodyLocationEntries(referral: Referral) {
```

Comment:
To avoid any error during this execution and to reduce code size here, could this be moved into an `ApplicationListener` please?